src/Workspaces/Core/Portable/Shared/Collections/IntervalTree`1.cs

```diff
@@ -309,17 +313,51 @@ public IEnumerator<T> GetEnumerator()
                 yield break;
             }
 
+            // The bool indicates if this is the first time we are seeing the node.
+            var candidates = new Stack<ValueTuple<Node,bool>>();
+            candidates.Push(ValueTuple.Create(root, true));
+            while (candidates.Count != 0)
+            {
+                var currentTuple = candidates.Pop();
+                var currentNode = currentTuple.Item1;
+                if (currentNode != null)
+                {
+                    if (currentTuple.Item2)
+                    {
+                        // First time seeing this node.  Mark that we've been seen and recurse
+                        // down the left side.  The next time we see this node we'll yield it
+                        // out.
+                        candidates.Push(ValueTuple.Create(currentNode.Right, true));
+                        candidates.Push(ValueTuple.Create(currentNode, false));
+                        candidates.Push(ValueTuple.Create(currentNode.Left, true));
+                    }
+                    else
+                    {
+                        yield return currentNode.Value;
+                    }
+                }
+            }
+        }
+
+        // For testing purposes only.  Allows a test to verify the actual AVL structure of the tree.
```

Comment:
Your thoughts match what I was thinking -- no single ordering is more correct than another. So I guess just validate with the returned order (which is the 'real' behavior anyways -- arguably you don't care if the rotate is wrong as long as the order comes out right :smile:).